### PR TITLE
10 add hosp admissions only model functionality

### LINKED ISSUE #10: Add hospital admissions only mode functionality
# Goal 
It is useful to compare the forecasts from the same model with and without the wastewater component included. Ideally, this doesn't require using the separate stan file and stan data object, but just turning off the wastewater component on the likelihood and making sure that the IHR is not time-varying

# Requirements
- [ ] implement a hospital admissions only renewal fit
- [ ] demonstrate the fit to only hospital admissions data in the getting started vignette

### PR DESCRIPTION
**Pull request by kaitejohnson**
This PR closes #10 and also has a bit of scope creep to fix the rendering of the vignette/some bugs in the figure generating function. Changes include:
- adding a component to the vignette that fits the model to only hospital admissions and plots the resulting hospital admissions forecasts
- adjusted the generate simulated data function and resulting package data to include a larger population size, bc the low counts of admissions were causing diagnostic issues
- modified `get_model_spec()` and `get_stan_data()` to take in `include_ww` as an argument, to specify whether or not to fit the model to wastewater data
- adjusted the way the figures were rendering in the vignette (mirroring [epinowcast vignette](https://github.com/epinowcast/epinowcast/blob/d957b21af07c7708d8ae28fc248fffe39d453828/vignettes/epinowcast.Rmd#L299C1-L302C7) to point to a figure folder which contains figures that render the width of the screen

### CODE REVIEW

**Comment by kaitejohnson** [comment]
Note this doesn't do what I mentioned in the MSR community meeting today -- it doesn't add the functionality to not have wastewater data. I will do this in a separate issue... I think by making an empty dataset

**Comment by kaitejohnson** [comment]
@dylanhmorris not a huge rush but this is ready for review and is relatively small (tweaks to vignette + the easy version of hospital admissions only functionality)

**Comment by kaitejohnson** [comment]
Although I am realizing now that I set up the github branch protections to require status checks to pass, which includes deploy of `pkgdown` but I think deploy wouldn't happen until the PR gets merged to main so maybe I should remove that one

**Comment by dylanhmorris** [comment]
> Although I am realizing now that I set up the github branch protections to require status checks to pass, which includes deploy of pkgdown but I think deploy wouldn't happen until the PR gets merged to main so maybe I should remove that one

Correct you should require a successful `build` but not a successful `deploy`)

**Comment by kaitejohnson** [comment]
> > Although I am realizing now that I set up the github branch protections to require status checks to pass, which includes deploy of pkgdown but I think deploy wouldn't happen until the PR gets merged to main so maybe I should remove that one
> 
> Correct you should require a successful `build` but not a successful `deploy`

@dylanhmorris  Would you mind taking a look at this and the ruleset? I tried to specify just `pkgdown / build (pull_request)` in the require status checks before passing but for some reason it still says expected, status waiting to be reported despite the successful build above

**Review by dylanhmorris — CHANGES_REQUESTED** [review]
Looks largely good, but there are a couple typos that will cause bugs and therefore need to be fixed.

File: R/wwinference.R [hunk 1]
```diff
@@ -84,6 +84,7 @@ wwinference <- function(ww_data,
     inf_to_count_delay = model_spec$inf_to_count_delay,
     infection_feedback_pmf = model_spec$infection_feedback_pmf,
     params = model_spec$params,
+    include_ww = as.numeric(model_spec$include_ww),
```
[the remark this answers is not in the document]
Comment (kaitejohnson) [reply R/wwinference.R:87]: It starts as a boolean, but it needs to be coerced to a 0 or 1 to get passed to stan.

File: R/figures.R [hunk 2]
```diff
@@ -24,11 +28,16 @@ get_plot_forecasted_counts <- function(draws,
...
                                        forecast_date,
-                                       count_type = "hospital admissions") {
-  p <- ggplot(draws |> dplyr::filter(
+                                       count_type = "hospital admissions",
+                                       n_draws_to_plot = 100) {
+  sampled_draws <- sample(1:max(draws$draw), n_draws_to_plot)
+
+  draws_to_plot <- draws |> dplyr::filter(
     name == "pred_counts",
     draw %in% sampled_draws
```
Comment (dylanhmorris) [inline R/figures.R:37]:
```suggestion
    draw %in% !!sampled_draws
```

File: R/figures.R [hunk 3]
```diff
@@ -67,23 +76,33 @@ get_plot_forecasted_counts <- function(draws,
...
-    draw %in% sampled_draws
-  ) |>
+                             forecast_date,
+                             n_draws_to_plot = 100) {
+  sampled_draws <- sample(1:max(draws$draw), n_draws_to_plot)
+
+  draws_to_plot <- draws |>
+    dplyr::filter(
+      name == "pred_ww",
+      draw %in% sampled_draws
```
Comment (dylanhmorris) [inline R/figures.R:99]:
```suggestion
      draw %in% !!sampled_draws
```

File: R/figures.R [hunk 4]
```diff
@@ -126,21 +145,30 @@ get_plot_ww_conc <- function(draws,
...
-  # R(t) of the hypothetical state
-  p <- ggplot(draws |> dplyr::filter(
-    name == "global R(t)",
+                               forecast_date,
+                               n_draws_to_plot = 100) {
+  sampled_draws <- sample(1:max(draws$draw), n_draws_to_plot)
+
+  draws_to_plot <- draws |> dplyr::filter(
+    name == "pred_counts",
     draw %in% sampled_draws
```
Comment (dylanhmorris) [inline R/figures.R:167]:
```suggestion
    draw %in% !!sampled_draws
```

File: R/figures.R [hunk 5]
```diff
@@ -174,20 +202,29 @@ get_plot_global_rt <- function(draws,
...
-                               forecast_date) {
-  p <- ggplot(draws |> dplyr::filter(
-    name == "subpop R(t)",
+                               forecast_date,
+                               n_draws_to_plot = 100) {
+  sampled_draws <- sample(1:max(draws$draw), n_draws_to_plot)
+
+  draws_to_plot <- draws |> dplyr::filter(
+    name == "pred_counts",
     draw %in% sampled_draws
```
Comment (dylanhmorris) [inline R/figures.R:224]:
```suggestion
    draw %in% !!sampled_draws
```

**Review by dylanhmorris — CHANGES_REQUESTED** [review]
a few more tidyeval things, otherwise good.

File: R/figures.R [hunk 6]
```diff
@@ -129,13 +139,18 @@ get_plot_ww_conc <- function(draws,
...
 #'
 #' @return A ggplot object containing the posterior draws of the global R(t)
 #' estimate
 #' @export
 #'
 get_plot_global_rt <- function(draws,
-                               forecast_date) {
+                               forecast_date,
+                               n_draws_to_plot = 100) {
+  sampled_draws <- sample(1:max(draws_df$draw), n_draws_to_plot)
```
[the remark this answers is not in the document]
Comment (dylanhmorris) [reply R/figures.R]: ^ Can be a separate PR but this should happen @kaitejohnson

**Review by dylanhmorris — APPROVED** [review]
LGTM

**Pull request merged**